nautobot_secrets_providers/providers/thycotic.py

```diff
@@ -0,0 +1,247 @@
+        # `required_settings` for the plugin config.
+        plugin_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+        if "thycotic" not in plugin_settings:
+            raise exceptions.SecretProviderError(secret, cls, "Thycotic Secret Server is not configured!")
+
+        # Try to get parameters and error out early.
+        parameters = secret.rendered_parameters(obj=obj)
+        try:
+            if "secret_id" in parameters.keys():
+                secret_id = parameters["secret_id"]
+            else:
+                secret_id = None
+            if "secret_path" in parameters.keys():
+                secret_path = parameters["secret_path"]
+            else:
+                secret_path = None
+            secret_selected_value = parameters["secret_selected_value"]
+        except KeyError as err:
+            msg = f"The secret parameter could not be retrieved for field {err}"
+            raise exceptions.SecretParametersError(secret, cls, msg) from err
+
+        if secret_id is None and secret_path is None:
+            msg = "The secret parameter could not be retrieved for field!"
+            raise exceptions.SecretParametersError(secret, cls, msg)
+
+        thycotic_settings = plugin_settings["thycotic"]
+        is_valid_base_url = "base_url" in thycotic_settings and thycotic_settings["base_url"] != ""
+        if not is_valid_base_url:
+            raise exceptions.SecretProviderError(secret, cls, "Thycotic Secret Server is not configured!")
+        is_valid_domain = "domain" in thycotic_settings and thycotic_settings["domain"] != ""
+        is_valid_password = "password" in thycotic_settings and thycotic_settings["password"] != ""
+        is_valid_tenant = "tenant" in thycotic_settings and thycotic_settings["tenant"] != ""
+        is_valid_token = "token" in thycotic_settings and thycotic_settings["token"] != ""
+        is_valid_username = "username" in thycotic_settings and thycotic_settings["username"] != ""
+        is_valid_ca_bundle_path = "ca_bundle_path" in thycotic_settings and thycotic_settings["ca_bundle_path"] != ""
+
+        return cls.query_thycotic_secret_server(
+            secret=secret,
+            base_url=thycotic_settings["base_url"],
+            ca_bundle_path=thycotic_settings["ca_bundle_path"] if is_valid_ca_bundle_path else None,
+            cloud_based=thycotic_settings["cloud_based"],
+            domain=thycotic_settings["domain"] if is_valid_domain else None,
+            password=thycotic_settings["password"] if is_valid_password else None,
+            secret_id=secret_id,
+            secret_path=secret_path,
+            secret_selected_value=secret_selected_value,
+            tenant=thycotic_settings["tenant"] if is_valid_tenant else None,
+            token=thycotic_settings["token"] if is_valid_token else None,
+            username=thycotic_settings["username"] if is_valid_username else None,
+            caller_class=cls,
```

Comment:
It appears to me that since you're just checking for the presence of a key in `thycotic_settings` for each of these `is_valid_foo` variables, and then passing them using the ternary operator to `ls.query_thycotic_secret_server()` as `get the key from the dict else None`, this could be simplified for readability and maintainability.

For example, instead of:

```python
        is_valid_base_url = "base_url" in thycotic_settings and thycotic_settings["base_url"] != ""
```

You could do:

```python
        base_url = thycotic_settings.get("base_url", None)
        if base_url is None:
            raise exceptions.SecretProviderError(secret, cls, "Thycotic Secret Server is not configured!")
```

When you pass it to `cls.query_thycotic_secret_server()`, instead you could just pass `base_url=base_url`.

For the rest of them, the same is also true. For example for `domain`. This:

```python
        is_valid_domain = "domain" in thycotic_settings and thycotic_settings["domain"] != ""
        # ... 
        domain=thycotic_settings["domain"] if is_valid_domain else None,
```

Could be simplified to:

```python
        domain = thycotic_settings.get("domain", None)
        # ... 
        domain=domain,
```

And then you could repeat this for the remaining arguments you're pulling in from `thycotic_settings`.

Reply:
```
is_valid_base_url = "base_url" in thycotic_settings and thycotic_settings["base_url"] != ""
if not is_valid_base_url:
    ....
```

I changed the documentation to use `None` as default value instead of `""`.

```
PLUGINS_CONFIG = {
    "nautobot_secrets_providers": {
        "thycotic": {  # https://github.com/thycotic/python-tss-sdk
            "base_url": os.getenv("SECRET_SERVER_BASE_URL", None),
            "ca_bundle_path": os.getenv("REQUESTS_CA_BUNDLE", None),
            ...
```

This will allow to use the suggested code